src/synapse/azext_synapse/_params.py

```diff
@@ -3,159 +3,204 @@
-            c.argument('detailed', action='store_true',
-                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+            c.argument('spark_pool_name', help='The name of the Spark pool.')
 
+    for scope in ['synapse spark job', 'synapse spark session']:
         with self.argument_context(scope + ' list') as c:
             c.argument('from_index', help='Optional parameter specifying which index the list should begin from.')
-            c.argument('detailed', action='store_true',
-                       help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
             c.argument('size',
                        help='The size of the returned list.By default it is 20 and that is the maximum.')
 
-    with self.argument_context('synapse spark batch create') as c:
-        c.argument('language', arg_type=get_enum_type(['SCALA', 'PYTHON', 'DOTNET'], default='SCALA'),
-                   help='The spark batch job language.')
+    with self.argument_context('synapse spark job submit') as c:
+        c.argument('main_definition_file', help='The main file used for the job.')
+        c.argument('main_class_name',
+                   help='The fully-qualified identifier or the main class that is in the main definition file.')
+        c.argument('command_line_arguments', nargs='+',
+                   help='Optional arguments to the job (Note: please use storage URIs for file arguments).')
+        c.argument('archives', nargs='+', help='The array of archives.')
+        c.argument('job_name', arg_type=name_type, help='The Spark job name.')
+        c.argument('reference_files', nargs='+',
+                   help='Additional files used for reference in the main definition file.')
+        c.argument('configuration', type=get_json_object, help='The configuration of Spark job.')
+        c.argument('executors', help='The number of executors.')
+        c.argument('executor_size', arg_type=get_enum_type(['Small', 'Medium', 'Large']), help='The executor size')
+        c.argument('tags', arg_type=tags_type)
+        c.argument('language', arg_type=get_enum_type(SparkBatchLanguage, default=SparkBatchLanguage.Scala),
+                   help='The Spark job language.')
 
     for scope in ['show', 'cancel']:
-        with self.argument_context('synapse spark batch ' + scope) as c:
-            c.argument('batch_id', options_list=['--id'], arg_group='Spark Batch',
-                       help='The id of the spark batch job.')
-
-    with self.argument_context('synapse spark batch show') as c:
-        c.argument('detailed', action='store_true',
-                   help='Optional query parameter specifying whether detailed response is returned beyond plain livy.')
+        with self.argument_context('synapse spark job ' + scope) as c:
+            c.argument('batch_id', options_list=['--livy-id'], arg_group='Spark job',

```

Comment:
>batch_id [](start = 24, length = 8)

job_id?

Reply:
This is per PM's requirement. Can we keep it as `--livy-id`?

---
In reply to: [424946588](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424946588) [](ancestors = 424946588)

Reply:
I mean change "batch_id" to "job_id" which is used in code but not exposed to CLI users, you can keep options_list=['--livy-id'] which is exposed to CLI users.

---
In reply to: [424997897](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r424997897) [](ancestors = 424997897,424946588)

Reply:
fixed.

---
In reply to: [425009176](https://github.com/Azure/azure-cli-extensions/pull/1581#discussion_r425009176) [](ancestors = 425009176,424997897,424946588)